src/main/java/com/chessmates/repository/RdsQueryExecutor.groovy

```diff
@@ -0,0 +1,115 @@
+package com.chessmates.repository
+
+import groovy.json.JsonOutput
+import groovy.json.JsonSlurper
+import org.postgresql.util.PGobject
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Primary
+import org.springframework.stereotype.Component
+
+import java.sql.DriverManager
+import java.sql.ResultSet
+
+/**
+ * class that executes insert/select queries PostGreSql DB hosted on AWS
+ */
+
+@Component
+@Primary
+class RdsQueryExecutor implements QueryExecutor {
+
+    @Value('${chessmates.rds.dbName}')
+    private String dbName
+
+    @Value('${chessmates.rds.userName}')
+    private String userName
+
+    @Value('${chessmates.rds.password}')
+    private String password
+
+    @Value('${chessmates.rds.hostName}')
+    private String hostName
+
+    @Value('${chessmates.rds.port}')
+    private String port
+
+    @Override
+    void executeInsert(givenTableName, givenItems) {
+        def injectJson = {
+            paramIndex, jsonString, stmt ->
+                def jsonObject = new PGobject()
+                jsonObject.setType("jsonb")
+                jsonObject.setValue(jsonString)
+                stmt.setObject(paramIndex, jsonObject)
+        }
+
+        def prepareAndInvokeInsertStatement = {
+            conn ->
+                def valueString = ', (?)'.multiply(givenItems.size()).toString().substring(1)
+                def stmt = conn.prepareStatement("INSERT INTO public.${givenTableName}(data) VALUES ${valueString};")
+                givenItems.eachWithIndex { x, i -> injectJson ((i + 1), JsonOutput.toJson(x).toString(), stmt)}
+                def itemsInserted = stmt.executeUpdate()
+                [stmt, itemsInserted]
+        }
+
+        executeQuery(prepareAndInvokeInsertStatement)
+    }
+
+    @Override
+    def executeSelect(givenTableName, givenId = null) {
```

Comment:
I'd probably type this for documentation purposes as it is a public method. I've only really relied on weak typing where a plain object is being returned. Looks like it is a List right?